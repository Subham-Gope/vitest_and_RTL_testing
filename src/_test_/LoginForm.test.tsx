// import { describe, test, expect } from "vitest";
// import { render, screen } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
// import LoginForm from "../components/LoginForm";

// describe("Testing the LoginForm.tsx component", () => {
//   test("lets user type and submit", async () => {
//     const { container } = render(<LoginForm />);

//     expect(container).toMatchSnapshot();

//     const input = screen.getByPlaceholderText("username");
//     const button = screen.getByRole("button");

//     await userEvent.type(input, "Sigma Lord");
//     await userEvent.click(button);

//     const message = screen.getByText("Submission Complete");

//     expect(message).toBeInTheDocument();
//   });
// });
