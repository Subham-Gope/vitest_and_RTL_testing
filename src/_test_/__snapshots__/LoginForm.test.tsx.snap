// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the LoginForm.tsx component > lets user type and submit 1`] = `
<div>
  <div>
    <input
      placeholder="username"
      type="text"
    />
    <button>
      Submit
    </button>
  </div>
</div>
`;
